sphinx_markdown_parser/markdown_parser.py

```diff
@@ -354,15 +364,43 @@ def visit_em(self, node):
     def visit_br(self, node):
         return nodes.Text('\n')
 
+    # note: logic is based on CommonMarkParser.visit_link()
     def visit_a(self, node):
         reference = nodes.reference()
         href = node.attrib.pop('href', '')
-        try:
-            r = urllib.parse.urlparse(href)
-            if r.path.endswith(".md"):
-              href = urllib.parse.urlunparse(r._replace(path = r.path[:-3] + ".html"))
-        except:
-            pass
+        url_check = urllib.parse.urlparse(href)
```

Comment:
@wlupton the previous code added .html, is this redundant (does sphinx or other code add it automatically) or is the behaviour actually changed? Some of my projects depend on this behaviour.

Reply:
I just did a test with this input:

```
These should all work:

* See [this file](extensions.html) for more info
* See [this file](extensions.md) for more info
* See [this file](extensions) for more info
```

It's actually the first one that doesn't work (the other two do). It's not detected as a link at all. Is that expected?

But I guess this also needs testing where there's an explicit link to an HTML file outside the "sphinx domain"?